### PR TITLE
Update registry credentials FAQ

### PR DESCRIPTION
**Pull request by squaremo**
We have made some advances since this was written; in particular, flux will now find credentials attached to service accounts.

### CODE REVIEW

File: site/faq.md [hunk 1]
```diff
@@ -101,21 +101,29 @@ system in order to read and update the manifests.
 
 ### How do I give Flux access to an image registry?
 
-Flux transparently looks at the image pull secret that you give for a
-workload, and thereby uses the same credentials that Kubernetes uses
-for pulling each image. If your pods are running, Kubernetes has
-pulled the images, and Flux should be able to access them.
+Flux transparently looks at the image pull secrets that you to
+workloads and service accounts, and thereby uses the same credentials
```
Comment (dholbach) [inline site/faq.md]: The first half of the sentence is a bit off. Can you rephrase to make it clearer?

File: site/faq.md [hunk 2]
```diff
@@ -101,21 +101,29 @@ system in order to read and update the manifests.
...
-   so Flux will work there too).
- - You can also attach image pull secrets to service accounts; Flux
-   does not at present try to obtain credentials via service accounts
-   (see
-   [weaveworks/flux#1043](https://github.com/weaveworks/flux/issues/1043)).
+   so Flux will work there too). See below regarding ECR.
+
+To work around the exceptional cases, you can mount a docker config into
+the Flux container. See the argument `--docker-config` in
+[the daemon arguments reference](https://github.com/weaveworks/flux/blob/master/site/daemon.md).
```
Comment (dholbach) [inline site/faq.md]: Maybe link directly to `daemon.md#flags`?

File: site/faq.md (hunk 1 above)
Comment (squaremo) [reply site/faq.md]: owps! corrected.

File: site/faq.md (hunk 2 above)
Comment (squaremo) [reply site/faq.md]: yup, good idea, ta.

File: site/faq.md [hunk 3]
```diff
@@ -117,7 +118,7 @@ There are exceptions:
 
 To work around the exceptional cases, you can mount a docker config into
 the Flux container. See the argument `--docker-config` in
-[the daemon arguments reference](https://github.com/weaveworks/flux/blob/master/site/daemon.md).
+[the daemon arguments reference](https://github.com/weaveworks/flux/blob/master/site/daemon.md#flags).
```
Comment (dholbach) [inline site/faq.md]: I think I had just used a relative link here, but no biggie. Let's get this landed.
